unpackaged/config/con_merge_testing/classes/REG_ContactMergeUniqueField_TEST.cls

```diff
@@ -0,0 +1,95 @@
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Contact Merge, Regression
+* @description Test method for testing unique fields merge on contact
+*/
+@isTest
+public class REG_ContactMergeUniqueField_TEST {
+    @isTest
+    public static  void shouldSuccessfullyMergeContactsWithLosingUniqueFieldSelected() {
+        String WINNING_UNIQUE_VALUE = 'Winning Unique Value';
+        String LOSING_UNIQUE_VALUE = 'Losing Unique Value';
+        String LOSING_LAST_NAME = '2Contact_forTests_Merge';
+
+        Contact winningContact = UTIL_UnitTestData_TEST.getContact();
+        winningContact.ContactMergeUniqueField__c = WINNING_UNIQUE_VALUE;
+
+        Contact losingContact = UTIL_UnitTestData_TEST.getContact();
+        losingContact.ContactMergeUniqueField__c = LOSING_UNIQUE_VALUE;
+        losingContact.LastName = LOSING_LAST_NAME;
+        
+        insert new Contact[] {winningContact, losingContact};
+
+        Test.setCurrentPageReference(new PageReference('Page.CON_ContactMerge'));
+        
+        CON_ContactMerge_CTRL controller = new CON_ContactMerge_CTRL(new ApexPages.Standardsetcontroller(new list<Contact>()));
+        controller.showContactSearchPage();
+        controller.searchText = CAO_Constants.CONTACT_FIRSTNAME_FOR_TESTS;
+        
+        List<Id> fixedSearchResults=new List<Id> {
+            winningContact.Id,
+            losingContact.Id
+        }; 
+
+        Test.setFixedSearchResults(fixedSearchResults); 
+
+        controller.search();
+        controller.searchResults[0].selected = true;
+        controller.searchResults[1].selected = true;
+        
+        controller.nextStep();
+        
+        CON_ContactMerge_TEST.selectMasterRecord(controller, winningContact.id);
+
+        List<CON_ContactMerge_CTRL.FieldRow> fieldRows = controller.fieldRows;  
+        for (CON_ContactMerge_CTRL.FieldRow row : fieldRows) {
+            if(row.fieldName == 'ContactMergeUniqueField__c') {
+                row.selectedValue = String.valueOf(losingContact.Id);
+            }
+            if(row.fieldName == 'LastName') {
+                row.selectedValue = String.valueOf(losingContact.Id);
+            }
+        }
+        
+        Test.startTest();
+        controller.mergeContacts();
+        Test.stopTest();
+        
+        Contact mergedContact = [SELECT ContactMergeUniqueField__c, LastName 
+          FROM Contact WHERE Id=:winningContact.Id];
+        System.assertEquals(LOSING_UNIQUE_VALUE, mergedContact.ContactMergeUniqueField__c,
```

Comment:
Since `mergeContacts()` is wrapped in a try/catch in the controller, should we add an assert here to confirm that no exception has been thrown with the unique field?

Reply:
add new assert to check if an error is thrown onto the page